packages/vm/src/imports.rs

```diff
@@ -95,156 +82,169 @@ mod errors {
-        Ok(data) => data,
-        Err(VmError::RegionLengthTooBig { .. }) => return errors::REGION_READ_LENGTH_TOO_BIG,
-        Err(_) => return errors::REGION_READ_UNKNOWN,
+/// This macro wraps the read_region function for the purposes of the functions below which need to report errors
+/// in its operation to the caller in the WASM runtime.
+/// On success, the read data is returned from the expression.
+/// On failure, an error number is wrapped in `Ok` and returned from the function.
+macro_rules! read_region {
+    ($ctx: expr, $ptr: expr, $length: expr) => {
+        match read_region($ctx, $ptr, $length) {
+            Ok(data) => data,
+            Err(err) => {
+                return Ok(match err {
+                    VmError::RegionLengthTooBig { .. } => errors::REGION_READ_LENGTH_TOO_BIG,
+                    _ => errors::REGION_READ_UNKNOWN,
+                })
+            }
+        }
     };
-    let value: Option<Vec<u8>> = match with_storage_from_context::<S, Q, _, _>(ctx, |store| {
-        store
-            .get(&key)
-            .or_else(|_| Err(make_backend_err("Error reading from backend")))
-    }) {
-        Ok(v) => v,
-        Err(VmError::UninitializedContextData { .. }) => return errors::NO_CONTEXT_DATA,
-        Err(_) => return errors::DB_UNKNOWN,
+}
+
+#[cfg(feature = "iterator")]
+/// This macro wraps the maybe_read_region function for the purposes of the functions below which need to report errors
+/// in its operation to the caller in the WASM runtime.
+/// On success, the optionally read data is returned from the expression.
+/// On failure, an error number is wrapped in `Ok` and returned from the function.
+macro_rules! maybe_read_region {
+    ($ctx: expr, $ptr: expr, $length: expr) => {
+        match maybe_read_region($ctx, $ptr, $length) {
+            Ok(data) => data,
+            Err(err) => {
+                return Ok(match err {
+                    VmError::RegionLengthTooBig { .. } => errors::REGION_READ_LENGTH_TOO_BIG,
+                    _ => errors::REGION_READ_UNKNOWN,
+                })
+            }
+        }
+// =======
+// /// Reads a storage entry from the VM's storage into Wasm memory
+// pub fn do_read<S: Storage, Q: Querier>(ctx: &mut Ctx, key_ptr: u32, value_ptr: u32) -> i32 {
+//     let key = match read_region(ctx, key_ptr, MAX_LENGTH_DB_KEY) {
+//         Ok(data) => data,
+//         Err(VmError::RegionLengthTooBig { .. }) => return errors::REGION_READ_LENGTH_TOO_BIG,
+//         Err(_) => return errors::REGION_READ_UNKNOWN,
+//     };
+//     let value: Option<Vec<u8>> = match with_storage_from_context::<S, Q, _, _>(ctx, |store| {
+//         store
+//             .get(&key)
+//             .or_else(|_| Err(make_backend_err("Error reading from backend")))
+//     }) {
+//         Ok(v) => v,
+//         Err(VmError::UninitializedContextData { .. }) => return errors::NO_CONTEXT_DATA,
+//         Err(_) => return errors::DB_UNKNOWN,
+// >>>>>>> origin/master
```

Comment:
What happened here? Some issue during merge?

Reply:
oops, forgot to delete that. i commented it out at some point and forgot to remove it